src/Engine/ProtoAssociative/CodeGen_SSA.cs

```diff
@@ -275,7 +275,7 @@ private List<AssociativeNode> BuildSSA(List<AssociativeNode> astList, ProtoCore.
                         BinaryExpressionNode bnode = (node as BinaryExpressionNode);
                         int generatedUID = ProtoCore.DSASM.Constants.kInvalidIndex;
 
-                        if (context.applySSATransform && core.Options.GenerateSSA)
+                        if (context.applySSATransform && core.Options.GenerateSSA && !bnode.IsInputExpression)
```

Comment:
Skip SSA for cases such as this:
```
a = null;
// update a
a = <some primitive>;
```

which would after SSA turn into this:
```
temp = null;
a = temp;
...
a = <some primitive>; // breaks!
```

Reply:
do you think this is a good constraint to have on SSA, or should we mark this as a temporary workaround and file a task to fix the optimization in the future?